autotst/conformer/systematic.py

```diff
@@ -306,43 +296,19 @@ def opt_conf(i, rmsd_cutoff):
   
         conformers[index] = copy_conf
 
-
-    processes = []
-    for i, conf in list(conformers.items()):
-        p = multiprocessing.Process(target=opt_conf, args=(i, rmsd_cutoff))
-        processes.append(p)
-
-    active_processes = []
-    for process in processes:
-        if len(active_processes) < multiprocessing.cpu_count():
-            process.start()
-            active_processes.append(process)
-            continue
-
-        else:
-            one_done = False
-            while not one_done:
-                for i, p in enumerate(active_processes):
-                    if not p.is_alive():
-                        one_done = True
-                        break
-
-            process.start()
-            active_processes[i] = process
-    complete = np.zeros_like(active_processes, dtype=bool)
-    while not np.all(complete):
-        for i, p in enumerate(active_processes):
-            if not p.is_alive():
-                complete[i] = True
+    num_threads = multiprocessing.cpu_count() - 1 or 1
+    pool = multiprocessing.Pool(processes=num_threads)
+    to_calculate_list = []
+    for i, conformer in list(conformers.items()):
+        to_calculate_list.append(conformer)
+    print(to_calculate_list)
+    results = pool.map(opt_conf,tuple(to_calculate_list))
+    pool.close()
+    pool.join()
 
     energies = []
-    for positions in list(return_dict.values()):
-        conf = conformer.copy()
-        conf.ase_molecule.positions = positions
-        conf.ase_molecule.set_calculator(calc)
-        energy = conf.ase_molecule.get_potential_energy()
-        conf.update_coords_from("ase")
-        energies.append((conf,energy))
+    for conformer in results:
+        energies.append((conformer,conformer.ase_molecule.get_potential_energy()))
```

Comment:
the energy is attached to the conformer in the `opt_conf` method as (`conformer.energy`).  I think we should replace `conformer.ase_molecule.get_potential_energy()` with `conformer.energy`, instead of calculating it again.  For the dftb calc, the `conformer.ase_molecule.get_potential_energy()` reads the energy from a dftb log, and the directory to look for the log is set in the `opt_conf` method.  So if the calc.directory doesn't change, this should work, but it might be safer to use `conformer.energy` instead

Reply:
Seconded.

Reply:
Seconded.